packages/devtools_app/lib/src/screens/inspector/inspector_controller.dart

```diff
@@ -194,46 +191,46 @@ class InspectorController extends DisposableController
   final bool isSummaryTree;
 
   /// Parent InspectorController if this is a details subtree.
-  InspectorController parent;
+  InspectorController? parent;
 
-  InspectorController details;
+  InspectorController? details;
 
   InspectorTreeController inspectorTree;
   final FlutterTreeType treeType;
 
   bool _disposed = false;
 
-  RateLimiter _refreshRateLimiter;
+  late RateLimiter _refreshRateLimiter;
 
   /// Groups used to manage and cancel requests to load data to display directly
   /// in the tree.
-  InspectorObjectGroupManager _treeGroups;
+  InspectorObjectGroupManager? _treeGroups;
```

Comment:
assigned to null in dispose

Reply:
for things that are only null due to being nulled in dispose, add a getter that is non-nullable to avoid polluting the code due to things nulled in dispose.

Reply:
They can turn out to be already nulls when a method is invoked. In this case usage of getter will be error prone.

So I prefer:
In methods, that assume the field is not null yet, to assign the field to local variable.
```
final treeGroupsLocal = _treeGroups!;
```

For methods that are ok with them to be null, to use null aware:
```
await _treeGroups?.pendingUpdateDone;
```

What do you think?